📖  Add extra information in the supported labels and annotations tables (#10425)
kubernetes-sigs/cluster-api: docs/book/src/reference/labels_and_annotations.md

| cluster.x-k8s.io/machine                                         | It is set on nodes identifying the machine the node belongs to. | Cluster API | Nodes |
| cluster.x-k8s.io/owner-kind                                      | It is set on nodes identifying the machine's owner kind the node belongs to.     | Cluster API | Nodes |
| cluster.x-k8s.io/owner-name                                      | It is set on nodes identifying the machine's owner name the node belongs to.    | Cluster API | Nodes |
| cluster.x-k8s.io/paused                                          | It can be applied to any Cluster API object to prevent a controller from processing a resource. Controllers working with Cluster API objects must check the existence of this annotation on the reconciled object.  | User | Any Cluster API resource |
| cluster.x-k8s.io/remediate-machine                               | It can be applied to a machine to manually mark it for remediation by MachineHealthCheck reconciler.   |    User          |    Machine         |

> || cluster.x-k8s.io/replicas-managed-by                             | It can be applied to MachinePool resources to signify that some external system is managing infrastructure scaling for that pool. See [the MachinePool documentation](../developer/architecture/controllers/machine-pool.md#externally-managed-autoscaler) for more details.  | Infrastructure Providers | MachinePool resources |

Reviewer: ```suggestion
| cluster.x-k8s.io/replicas-managed-by                             | It can be applied to MachinePool resources to signify that some external system is managing infrastructure scaling for that pool. See [the MachinePool documentation](../developer/architecture/controllers/machine-pool.md#externally-managed-autoscaler) for more details.  | Infrastructure Providers | MachinePool resources |
```
(the formatting seems of)

| controlplane.cluster.x-k8s.io/kubeadm-cluster-configuration      | It is a machine annotation that stores the json-marshalled string of KCP ClusterConfiguration. This annotation is used to detect any changes in ClusterConfiguration and trigger machine rollout in KCP.    | User | Machines |
| controlplane.cluster.x-k8s.io/remediation-for                    | It is a machine annotation that links a new machine to the unhealthy machine it is replacing.      | User | Machines |
| controlplane.cluster.x-k8s.io/remediation-in-progress            | It is a KCP annotation that tracks that the system is in between having deleted an unhealthy machine and recreating its replacement.     | User | Machines |
| controlplane.cluster.x-k8s.io/skip-coredns                       | It explicitly skips reconciling CoreDNS if set.   | User | Control plane machines |
| controlplane.cluster.x-k8s.io/skip-kube-proxy                    | It explicitly skips reconciling kube-proxy if set.  | User | Control plane machines |

> | machine.cluster.x-k8s.io/certificates-expiry                     | It captures the expiry date of the machine certificates in RFC3339 format. It is used to trigger rollout of control plane machines before certificates expire. It can be set on BootstrapConfig and Machine objects. The value set on Machine object takes precedence. The annotation is only used by control plane machines.      | User | BootstrapConfig and Machine objects |

Reviewer: ```suggestion
| machine.cluster.x-k8s.io/certificates-expiry                     | It captures the expiry date of the machine certificates in RFC3339 format. It is used to trigger rollout of control plane machines before certificates expire. It can be set on BootstrapConfig and Machine objects. The value set on Machine object takes precedence. The annotation is only used by control plane machines.      | Cluster API/User | BootstrapConfig/Machine |
```

| machine.cluster.x-k8s.io/exclude-node-draining                   | It explicitly skips node draining if set.    | User | Machines |
| machine.cluster.x-k8s.io/exclude-wait-for-node-volume-detach     | It explicitly skips the waiting for node volume detaching if set.      | User | Machines |

> | machinedeployment.clusters.x-k8s.io/desired-replicas             | It is the desired replicas for a machine deployment recorded as an annotation in its machine sets. Helps in separating scaling events from the rollout process and for determining if the new machine set for a deployment is really saturated.   |    Cluster API | Machine deployment's machine sets |

Reviewer: ```suggestion
| machinedeployment.clusters.x-k8s.io/desired-replicas             | It is the desired replicas for a machine deployment recorded as an annotation in its machine sets. Helps in separating scaling events from the rollout process and for determining if the new machine set for a deployment is really saturated.   |    Cluster API | MachineSets |
```

> | machinedeployment.clusters.x-k8s.io/max-replicas                 | It is the maximum replicas a deployment can have at a given point, which is machinedeployment.spec.replicas + maxSurge. Used by the underlying machine sets to estimate their proportions in case the deployment has surge replicas.  |     Cluster API    | Machine deployment's machine sets |

Reviewer: ```suggestion
| machinedeployment.clusters.x-k8s.io/max-replicas                 | It is the maximum replicas a deployment can have at a given point, which is machinedeployment.spec.replicas + maxSurge. Used by the underlying machine sets to estimate their proportions in case the deployment has surge replicas.  |     Cluster API    | MachineSets |
```

> | machinedeployment.clusters.x-k8s.io/revision                     | It is the revision annotation of a machine deployment's machine sets which records its rollout sequence.         |         |              |

Reviewer: ```suggestion
| machinedeployment.clusters.x-k8s.io/revision                     | It is the revision annotation of a machine deployment's machine sets which records its rollout sequence.         |  Cluster API  | MachineSet |
```

> | machinedeployment.clusters.x-k8s.io/revision-history             | It maintains the history of all old revisions that a machine set has served for a machine deployment.   | Cluster API | Machine deployment's machine sets |

Reviewer: ```suggestion
| machinedeployment.clusters.x-k8s.io/revision-history             | It maintains the history of all old revisions that a machine set has served for a machine deployment.   | Cluster API | MachineSets |
```

| pre-drain.delete.hook.machine.cluster.x-k8s.io                   | It specifies the prefix we search each annotation for during the pre-drain.delete lifecycle hook to pause reconciliation of deletion. These hooks will prevent removal of draining the associated node until all are removed.   | User | Machines |
| pre-terminate.delete.hook.machine.cluster.x-k8s.io               | It specifies the prefix we search each annotation for during the pre-terminate.delete lifecycle hook to pause reconciliation of deletion. These hooks will prevent removal of an instance from an infrastructure provider until all are removed.   | User | Machines |

> | topology.cluster.x-k8s.io/defer-upgrade                          | It can be used to defer the Kubernetes upgrade of a single MachineDeployment topology. If the annotation is set on a MachineDeployment topology in Cluster.spec.topology.workers, the Kubernetes upgrade for this MachineDeployment topology is deferred. It doesn't affect other MachineDeployment topologies.                                                                                               | Cluster API |	MachineDeployment topology |

Reviewer: ```suggestion
| topology.cluster.x-k8s.io/defer-upgrade                          | It can be used to defer the Kubernetes upgrade of a single MachineDeployment topology. If the annotation is set on a MachineDeployment topology in Cluster.spec.topology.workers, the Kubernetes upgrade for this MachineDeployment topology is deferred. It doesn't affect other MachineDeployment topologies.                                                                                               | Cluster API |	MachineDeployments in Cluster.topology |
```

| topology.cluster.x-k8s.io/dry-run	                               | It is an annotation that gets set on objects by the topology controller only during a server side dry run apply operation. It is used for validating update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate). When the annotation is set and the admission request is a dry run, the webhook should deny validation due to immutability. By that the request will succeed (without any changes to the actual object because it is a dry run) and the topology controller will receive the resulting object. |	Cluster API	| Template rotation objects |

> | topology.cluster.x-k8s.io/hold-upgrade-sequence	               | It can be used to hold the entire MachineDeployment upgrade sequence. If the annotation is set on a MachineDeployment topology in Cluster.spec.topology.workers, the Kubernetes upgrade for this MachineDeployment topology and all subsequent ones is deferred.	| Cluster API	| MachineDeployment topology |

Reviewer: ```suggestion
| topology.cluster.x-k8s.io/hold-upgrade-sequence	               | It can be used to hold the entire MachineDeployment upgrade sequence. If the annotation is set on a MachineDeployment topology in Cluster.spec.topology.workers, the Kubernetes upgrade for this MachineDeployment topology and all subsequent ones is deferred.	| Cluster API	| MachineDeployments in Cluster.topology |
```

> | topology.cluster.x-k8s.io/upgrade-concurrency	                   | It can be used to configure the maximum concurrency while upgrading MachineDeployments of a classy Cluster. It is set as a top level annotation on the Cluster object. The value should be >= 1. If unspecified the upgrade concurrency will default to 1. |	Cluster API | Cluster object |

Reviewer: ```suggestion
| topology.cluster.x-k8s.io/upgrade-concurrency	                   | It can be used to configure the maximum concurrency while upgrading MachineDeployments of a classy Cluster. It is set as a top level annotation on the Cluster object. The value should be >= 1. If unspecified the upgrade concurrency will default to 1. |	Cluster API | Cluster |
```

> | unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check | It can be used to disable the webhook check on update that disallows a pre-existing Cluster to be populated with Topology information and Class. | User     |   Cluster objects      |

Reviewer: ```suggestion
| unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check | It can be used to disable the webhook check on update that disallows a pre-existing Cluster to be populated with Topology information and Class. | User     |   Cluster      |
```

> | unsafe.topology.cluster.x-k8s.io/disable-update-version-check    | It can be used to disable the webhook checks on update that disallows updating the .topology.spec.version on certain conditions. | User | Cluster objects |

Reviewer: ```suggestion
| unsafe.topology.cluster.x-k8s.io/disable-update-version-check    | It can be used to disable the webhook checks on update that disallows updating the .topology.spec.version on certain conditions. | User | Cluster |
```
